fuse_constraints/include/fuse_constraints/absolute_pose_2d_stamped_constraint.h

```diff
@@ -88,21 +98,21 @@ class AbsolutePose2DStampedConstraint : public fuse_core::Constraint
    *
    * Order is (x, y, yaw)
    */
-  const fuse_core::Vector3d& mean() const { return mean_; }
+  const fuse_core::VectorXd& mean() const { return mean_; }
 
   /**
    * @brief Read-only access to the square root information matrix.
    *
    * Order is (x, y, yaw)
    */
-  const fuse_core::Matrix3d& sqrtInformation() const { return sqrt_information_; }
+  const fuse_core::MatrixXd& sqrtInformation() const { return sqrt_information_; }
 
   /**
    * @brief Compute the measurement covariance matrix.
    *
    * Order is (x, y, yaw)
    */
-  fuse_core::Matrix3d covariance() const { return (sqrt_information_.transpose() * sqrt_information_).inverse(); }
+  fuse_core::MatrixXd covariance() const { return (sqrt_information_.transpose() * sqrt_information_).inverse(); }
```

Comment:
I suspect you will have problems here. The sqrt_information_ matrix is potentially non-square, which will lead to a singular info matrix (sqrt_info^T * sqrt_info). That will make the inverse() call unhappy. You'll need to do some sort of pseudo-inverse. Luckily for you, I worked that out already. :)

https://github.com/locusrobotics/fuse/blob/devel/fuse_constraints/include/fuse_constraints/absolute_constraint_impl.h#L92